tests/integration/ui_write/test_scratch.py

```diff
@@ -21,34 +20,65 @@ def getContainer(self, nwbfile):
         return nwbfile.get_scratch('foo', convert=False)
 
     def roundtripScratch(self, data, case):
-        filename = 'test_scratch_%s.nwb' % case
+        self.filename = 'test_scratch_%s.nwb' % case
         description = 'a file to test writing and reading a scratch data of type %s' % case
         identifier = 'TEST_scratch_%s' % case
         nwbfile = NWBFile(description, identifier, self.start_time, file_create_date=self.create_date)
         nwbfile.add_scratch(data, name='foo', notes='test scratch')
 
-        writer = NWBHDF5IO(filename, mode='w')
-        writer.write(nwbfile)
-        writer.close()
-        reader = NWBHDF5IO(filename, mode='r')
-        read_nwbfile = reader.read()
+        self.writer = NWBHDF5IO(self.filename, mode='w')
+        self.writer.write(nwbfile)
+        self.writer.close()
 
-        ret = read_nwbfile.get_scratch('foo')
-        reader.close()
-        return ret
+        self.reader = NWBHDF5IO(self.filename, mode='r')
+        self.read_nwbfile = self.reader.read()
+        return self.read_nwbfile.get_scratch('foo')
 
     def test_scratch_convert_list(self):
         data = [1, 2, 3, 4]
         ret = self.roundtripScratch(data, 'list')
         assert_array_equal(data, ret)
+        self.validate()
 
     def test_scratch_convert_ndarray(self):
         data = np.array([1, 2, 3, 4])
         ret = self.roundtripScratch(data, 'ndarray')
         assert_array_equal(data, ret)
+        self.validate()
 
     def test_scratch_convert_DataFrame(self):
         data = pd.DataFrame(data={'col1': [1, 2, 3, 4], 'col2': ['a', 'b', 'c', 'd']})
-        ret = self.roundtripScratch(data, 'DataFrame')
+        warn_msg = 'Notes argument is ignored when adding a pandas DataFrame to scratch'
+        with self.assertWarnsRegex(UserWarning, warn_msg):
+            ret = self.roundtripScratch(data, 'DataFrame')
         assert_array_equal(data.values, ret.values)
         assert_array_equal(data.index.values, ret.index.values)
+        self.validate()
+
+    def test_scratch_container(self, validate=True, **kwargs):
```

Comment:
Can you make this a camel case method, and then have `test_*` method that calls the camel case method? That way `test_scratch_container_with_name` and `test_scratch_container_with_notes` don't depend on another test

Reply:
Instead of a camel case method, how about prependign the name with an underscore?

Reply:
That works just fine